### PR TITLE
Remove GMV new

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR moves the explicit part of the grid mean equations to the explicit part of the implicit solve--allowing us to remove the grid mean `new` property. This PR also inlines some functions to prepare for more loop fusing.

### CODE REVIEW

File: src/Turbulence_PrognosticTKE.jl [hunk 1]
```diff
@@ -500,15 +500,15 @@ function update(edmf::EDMF_PrognosticTKE, GMV::GridMeanVariables, Case::CasesBas
     implicit_eqs.A_uv_gm .= construct_tridiag_diffusion_gm(grid, TS.dt, edmf.rho_ae_KM, ref_state.rho0_half, edmf.ae)
     Δzi = grid.Δzi
     @inbounds for k in real_center_indices(grid)
-        implicit_eqs.b_u_gm[k] = GMV.U.values[k]
-        implicit_eqs.b_v_gm[k] = GMV.V.values[k]
-        implicit_eqs.b_q_tot_gm[k] = edmf.EnvVar.QT.values[k]
```
Comment (ilopezgp) [inline src/Turbulence_PrognosticTKE.jl:505]: Was this an error before, where we were taking the environmental values?

File: src/Turbulence_PrognosticTKE.jl [hunk 2]
```diff
@@ -500,15 +500,15 @@ function update(edmf::EDMF_PrognosticTKE, GMV::GridMeanVariables, Case::CasesBas
...
-        implicit_eqs.b_θ_liq_ice_gm[k] = edmf.EnvVar.H.values[k]
+        implicit_eqs.b_u_gm[k] = GMV.U.values[k] + TS.dt * GMV.U.tendencies[k]
+        implicit_eqs.b_v_gm[k] = GMV.V.values[k] + TS.dt * GMV.V.tendencies[k]
```
Comment (ilopezgp) [inline src/Turbulence_PrognosticTKE.jl:504]: So we are computing the implicit tendency in here in terms of an *intermediate step* ?

**Comment by ilopezgp** [comment]
Let's document the reasons behind this clearly before changing the solver. I am not sure the proposed solution is correct.

**Comment by charleskawczynski** [comment]
Superseded by #343

**Pull request closed without merging**